sw/device/lib/base/libgcc_polyfill_unittest.cc

```diff
@@ -0,0 +1,117 @@
+
+#include <ostream>
+#include <stdint.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+// Reach into bitfield.c and grab the internal testing symbols.
+extern "C" {
+int32_t libgcc_bswap_i32(int32_t val);
+int libgcc_popcount_i32(int32_t val);
+}
+
+namespace math_unittest {
+namespace {
+
+struct Bswap32Vector {
+  uint32_t a, b;
+  friend void operator<<(std::ostream &out, const Bswap32Vector &v) {
+    out << "{" << v.a << ", " << v.b << "}";
+  }
+};
+
+class Bswap32Test : public testing::TestWithParam<Bswap32Vector> {};
+
+TEST_P(Bswap32Test, Bswap32) {
+  EXPECT_EQ(libgcc_bswap_i32(GetParam().a), GetParam().b);
+}
+
+// clang-format off
+// Simple python snippet for generating vectors:
+//
+// import random
+// def bswap(x):
+//   return int.from_bytes(x.to_bytes(4, byteorder='little'), byteorder='big', signed=False)
+// for _ in range(0, 16):
+//   a1 = random.getrandbits(32)
+//   a2 = random.getrandbits(32)
+//   a3 = random.getrandbits(32)
+//   a4 = random.getrandbits(32)
+//   print(f"{{{a1}u, {bswap(a1)}u}}, {{{a2}u, {bswap(a2)}u}}, {{{a3}u, {bswap(a3)}u}}, {{{a4}u, {bswap(a4)}u}},")
+constexpr Bswap32Vector kBswap32Vectors[] = {
+  {175026543u, 1873899018u}, {2905347885u, 756493485u}, {4149642148u, 2760070903u}, {2246878737u, 296152197u},
+  {1929500587u, 2882994547u}, {2058127287u, 3079122042u}, {3946287479u, 2006005739u}, {51404757u, 3579777027u},
+  {1896150144u, 2163475569u}, {335187835u, 2073033235u}, {1045084252u, 1555581502u}, {3482594570u, 170759375u},
+  {3068065440u, 2700533430u}, {1687989070u, 1319869540u}, {4217086075u, 2074106875u}, {1834686227u, 320297837u},
+  {1387040119u, 2005773394u}, {962246504u, 1756846649u}, {2147935717u, 3856991872u}, {3868729971u, 1931647206u},
+  {369394435u, 58917910u}, {2451638174u, 2651267474u}, {2314346717u, 3709923977u}, {3164065408u, 2161022908u},
+  {68746954u, 3405649924u}, {953091349u, 352702264u}, {1015599226u, 2060486716u}, {3780058840u, 3625865185u},
+  {1482255209u, 1768642904u}, {1289715167u, 3749568332u}, {1208370945u, 21431880u}, {884236658u, 1919005748u},
+  {1251138604u, 752652874u}, {2746512660u, 343258275u}, {3408258377u, 1239754187u}, {1941160821u, 1975759731u},
+  {3601250101u, 901228246u}, {552045337u, 428599072u}, {525725375u, 3220067615u}, {1493075448u, 4169530968u},
+  {458834222u, 776034587u}, {1725702518u, 1981930598u}, {2658362673u, 828732318u}, {694561013u, 4113065513u},
+  {2234693117u, 4256510597u}, {2146340768u, 2693787263u}, {3036393057u, 1638857652u}, {515412848u, 1888729118u},
+  {1484651919u, 2415754584u}, {851907104u, 538101554u}, {1505051753u, 1766110553u}, {3408912721u, 1373712331u},
+  {1695131141u, 94767461u}, {3901279206u, 3872098536u}, {1644878809u, 3655010914u}, {3368358259u, 1930544584u},
+  {1094569037u, 1305230657u}, {215149339u, 468439564u}, {3955150114u, 583909099u}, {2147398144u, 11730559u},
+  {2508623926u, 915441301u}, {2924276618u, 2330676398u}, {3742087918u, 4005170143u}, {2271260870u, 3333447815u},
```

Comment:
nit: If these were expressed in hex it would be clearer that/why they were correct